projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/credentials/JobCredentialsService.java

```diff
@@ -54,11 +56,13 @@ public void createJobCredentials(String jobName) {
     String principal = getJobPrincipalName(jobName);
     File keytabFile;
     try {
+      // create a temp file name by creating a temp file and then deleteing the actual file.
       keytabFile = File.createTempFile(principal, ".keytab");
+      assertTrue(keytabFile.delete());
```

Comment:
I assume this raises an AssertionError which is not good practice in production code (though they are arguments for both sides). In this project we avoid them. 

Logging a warning message may be more suitable here ?

Reply:
Change made. 
I think a log line would be excessive as it should never happen.